shop/shop/src/main/java/ro/msg/learning/shop/Mapper.java

```diff
@@ -0,0 +1,43 @@
+package ro.msg.learning.shop;
+
+import org.springframework.stereotype.Component;
+import ro.msg.learning.shop.dto.ProductCategoryDto;
+import ro.msg.learning.shop.dto.ProductDto;
+import ro.msg.learning.shop.dto.SupplierDto;
+import ro.msg.learning.shop.model.Product;
+import ro.msg.learning.shop.model.ProductCategory;
+import ro.msg.learning.shop.model.Supplier;
+import java.math.BigDecimal;
+
+@Component
+public
+class Mapper {
+    public ProductDto toDto(Product product) {
+        String name = product.getName();
+        Integer id = product.getId();
+        Integer categoryId = product.getCategory().getId();
+        Integer supplierId = product.getSupplier().getId();
+        String description = product.getDescription();
+        BigDecimal price = product.getPrice();
+        Double weight = product.getWeight();
+        String categoryName = product.getCategory().getName();
+        String categoryDescription = product.getCategory().getDescription();
+        String supplierName = product.getSupplier().getName();
+        String imageUrl = product.getImageUrl();
+        ProductCategoryDto categoryDto = new ProductCategoryDto(categoryId, product.getCategory().getName(), product.getCategory().getDescription());
+        SupplierDto supplierDto = new SupplierDto(supplierId, product.getSupplier().getName());
+        return new ProductDto(id, categoryId, supplierId, name, description, price, weight, categoryDto, supplierDto, imageUrl);
+    }
+
+    public Product toProduct(ProductDto productDto) {
+        ProductCategory productCategory = new ProductCategory(productDto.getCategory().getName(), productDto.getCategory().getDescription());
+        productCategory.setId(productDto.getCategory().getId());
+
+        Supplier supplier = new Supplier(productDto.getSupplier().getName());
+        supplier.setId(productDto.getSupplier().getId());
+
+        Product product = new Product(productDto.getName(), productDto.getDescription(), productDto.getPrice(), productDto.getWeight(), productCategory, supplier, productDto.getImageUrl());
```

Comment:
In order to reduce the number of parameters (recommended is max of 3, sometimes 5) you can send the whole `productDto` and retrieve the attributes within the constructor